### ISSUE TITLE
DaPars_Extract_Anno.py - coordinates in output BED file (and 'Loci' column) are shifted 1 nucleotide from source transcript

### ISSUE THREAD

**Issue by SamBryce-Smith** [ZhengXia/dapars#16]
TL;DR:

**\+ strand: DaPars BED/'Loci' 3'UTR start is 1 nt downstream of the actual annotated start**
**\- strand: Dapars BED/'Loci' 3'end of the 3'UTR is 1 nt upstream of the actual annotated end**

In [lines 38-45](https://github.com/ZhengXia/dapars/blob/546bdd4d809b097fd7a946d42b56e0efa0228472/src/DaPars_Extract_Anno.py#L38) of DaPars_Extract_Anno.py, the 'start' coordinates for UTRs (last exons) are converted to '1-based' by adding 1 to the start coordinate. Given that the script outputs a BED file, and according to the [BED format convention start coordinates are included in the range](https://genome.ucsc.edu/FAQ/FAQformat.html#format1), adding 1 to the start coordinate actually excludes the first nucleotide of the UTR on the plus strand and the last nucleotide of the UTR on the minus strand.

Here is an example of extracted UTRs on the + strand. 'D Extracted 3'UTR' is the output BED file of `DaPars_Extract_Anno.py`, with 'D2 Extracted 3'UTR' the same file from the DaPars2 repo (the script is identical between the two tools). 'GTF' is the reference GTF file from which the input BED12 file was derived. In both cases, the last exon reported in the DaPars BED file begins 1nt downstream of the source transcript:

![image](https://user-images.githubusercontent.com/49978382/138076641-0f202af4-9394-44b9-9d92-eb13fdf4b23d.png)

Again, on the - strand the DaPars BED file terminates 1nt upstream of the source transcript end:

![image](https://user-images.githubusercontent.com/49978382/138078488-f224eb3d-02a5-4ad0-b7c7-e2a3f1e9c176.png)

I'm not sure how DaPars handles this internally (i.e. the coordinates in the BED file may not be interpreted according to BED conventions) and regardless I don't expect 1nt to make a drastic difference to the algorithm's output. **The real issue comes when using the BED file/'Loci' column in output files to extract predicted polyA sites**, especially the distal polyA site on the minus strand (as the Start coordinate which corresponds to the UTR end is shifted by 1nt). 

Screenshots were generated using output from the [DaPars](https://github.com/iRNA-COSI/APAeval/tree/24f27f42d9caad10890f8721ac3990a4a949a9e3) & [DaPars2](https://github.com/iRNA-COSI/APAeval/tree/a9d86ac37f1b9f4c286871489048cee8edbf3b98) execution workflows from the APAeval project with [test data](https://github.com/iRNA-COSI/APAeval/tree/main/tests/test_data) used as input. 

Edit: This issue has also been reported to the DaPars2 repo at 3UTR/DaPars2#8
